### ISSUE TITLE
Custom multivariate transformation and sklearn2pmml

### ISSUE THREAD

**Issue by jwang2sc** [jpmml/sklearn2pmml#51]
suppose that I have a data set with two features and one response variable like following:

"height", "width", "is_palm_tree",

in which the "height" and "width" variables are predictors and the "is_palm_tree" is training label for our classifier. We'd like to compute a derived variable like 

ht_wt_ratio = height/width,
 and build a logistic regression classifier using "ht_wt_ratio" to classify "is_palm_tree". We'd like to output a PMML file with the custom transformation and classifier. Is that something we can do with sklearn transformers and sklearn2pmml ?

**Comment by vruusmann** [comment]
At the moment you'd need to create a custom `com.mycompany.Divider` transformation. The [SkLearn2PMML-Plugin](https://github.com/jpmml/sklearn2pmml-plugin) project provides "rails" for that - shouldn't take more than an hour or two.

In the long run, there should be some sort of `sklearn2pmml.preprocessing.ExpressionTransformer` transformation for doing simple math work. You would write the expression as a string, which would be `eval()`-ed on the Python side, and be parsed/translated on the PMML/Java side:
```Python
expr = ExpressionTransformer("X[0] / X[1]")
Xt = expr.fit_transform(X)
```

**Comment by vruusmann** [comment]
I've started work on `sklearn2pmml.preprocessing.ExpressionTransformer`, and the MVP implementation of it should be released later this week.

The Python side of the custom transformer class is extremely simple:
```Python
class ExpressionTransformer(TransformerMixin):

	def __init__(self, expr):
		self.expr_ = expr

	def fit(self, X, y = None):
		return self

	def transform(self, X, y = None):
		return eval(self.expr_)
```

Sample usage:
```Python
transformer = ExpressionTransformer("X[:, 0] / X[:, 1]")
X = numpy.array([[0.5, 0.5], [1.0, 2.0]])
Xt = transformer.fit_transform(X)
```

Initially, the PMML/Java side will be limited to the four basic arithmetic operations `+`, `-`, `*` and `/`. But that should be enough to satisfy this feature request.

**Comment by jwang2sc** [comment]
thank you @vruusmann for implementing this feature. I have a slightly more complicated use case than simple ratio, I'd need to have a transformation similar to empirical bayes 

eb_estimator = (numerator + alpha) / (denom + alpha + beta),

can I directly write something like 
ExpressionTransformer("(X[:, 0] + 2) / (X[:, 1] + 4)"),
or I need to implement this in multiple steps? will check back with you when the feature is about to come out. :)

**Comment by vruusmann** [comment]
I'll take your requirements into consideration - will add support for parenthesized expressions and numeric constants (eg. `2` and `4`) as well.

**Comment by vruusmann** [comment]
Class `sklearn2pmml.preprocessing.ExpressionTransformer` is available starting from SkLearn2PMML version 0.22.0 - please upgrade!

Example usage:
https://github.com/jpmml/jpmml-sklearn/blob/master/src/test/resources/main.py#L353

@jwang2sc Could you drop me an e-mail (see my GitHub profile)? I've got an idea that I'd like to share.
